src/game/World.cpp

```diff
@@ -2244,6 +2244,45 @@ void World::SendWorldText(int32 string_id, ...)
     va_end(ap);
 }
 
+// Send a System Message to all players in the same battleground or queue (except self if mentioned)
+void World::SendWorldTextToBGAndQueue(int32 string_id, uint32 queuedPlayerLevel, BattleGroundQueueTypeId queueType, ...)
+{
+    va_list ap;
+    va_start(ap, string_id);
+
+    MaNGOS::WorldWorldTextBuilder wt_builder(string_id, &ap);
+    MaNGOS::LocalizedPacketListDo<MaNGOS::WorldWorldTextBuilder> wt_do(wt_builder);
+    for (const auto& itr : m_sessions)
+    {
+        if (WorldSession* session = itr.second)
+        {
+            Player* player = session->GetPlayer();
+            if (player && player->IsInWorld())
+            {
+                // Always announce it to all GMs.
+                if (player->GetSession()->GetSecurity() > SEC_PLAYER)
```

Comment:
Actually you have the session above, dont need to call GetSession here, just use session->

Reply:
Done.